journal/journal-app.js

```diff
@@ -0,0 +1,147 @@
+window.addEventListener('DOMContentLoaded', init);
+
+// Main function that initializes every other function
+function init() {
+    const journalContainer = document.getElementById('journal-app');
+    const addButton = journalContainer.querySelector('.add-journal');
+    const overlayRef = document.getElementById('overlay');
+    const modalRef = document.getElementById('modal');
+
+    addButton.addEventListener('click', () => addJournalNew(journalContainer, false));
+
+    // When a user clicks "out" of the journal content/popup, it closes it
+    
+    overlayRef.addEventListener('click', () => {
+        overlayRef.style.display = 'none';
+        modalRef.style.display = 'none';
+    });
+
+    // Add saved journals from localStorage
+    addJournalNew(journalContainer, true);
+}
+
+// Display the new journal element to the page
+function addJournalNew(journalContainer, existing) {
+    
+    const journalList = getJournals();
+    let journalID = 0;
+
+    const journalTemplate = {
+        id: Math.floor(Math.random() * 2000000),
+        title: "",
+        documentation: "",
+        reflection: ""
+    };
+    
+    const modalRef = document.getElementById('modal');
+
+    if (existing) {
+
+        journalList.forEach(journal => {
+            journalID = createJournalElement(journal.id, journal.title, 
+                journal.documentation, journal.reflection, modalRef);
+
+                const journalElementTest = document.createElement('div');
+                journalElementTest.classList.add('journal-widget');
+                journalElementTest.textContent = 'Placeholder Title';
+                journalElementTest.setAttribute('widget-id', journalID);
+            
+                journalElementTest.addEventListener('dblclick', function(event) {
+                    document.getElementById('overlay').style.display = 'block';
+                    const currentJournalID = event.target.getAttribute('widget-id');
+                    const journalEntries = document.querySelectorAll('.journal-entry');
+                    hideOtherJournalEntries(journalEntries, currentJournalID);
+                    document.getElementById('modal').style.display = 'block';
+                });
+
+                journalContainer.append(journalElementTest);
+        });
+    }
+    else {
+        journalID = createJournalElement(journalTemplate.id, journalTemplate.title, 
+            journalTemplate.documentation, journalTemplate.reflection, modalRef);
+            
+            const journalElementTest = document.createElement('div');
```

Comment:
the `else` block has a bit of the repeating code as the block above. Consider simplifying. I know you mentioned sanjith has a more polished version of your code so feel free to ignore if he addressed this

Reply:
I'm aware that I essentially have copy-and-paste code for the `existing` code section. Like you said, Sanjith and James forked from my branch and extended/optimized it by creating a new function for the copy-and-paste code called `createJournalWidget()`.